frontend/catalyst/programs/verification.py

```diff
@@ -49,6 +57,39 @@ def _verify_nested(
     return state
 
 
+def _verify_observable(obs: Operation, _obs_checker: Callable) -> bool:
+    """Validates whether or not an observable is accepted by QJITDevice.
+
+    Args:
+        obs(Operator): the observable to be validated
+        _obs_checker(Callable): a callable that takes an observable
+            and raises an error if the observable is unsupported.
+
+    Raises:
+        DifferentiableCompileError: gradient-related error
+        CompileError: compilation error
+
+    If the observable is built on one or multiple other observables, check
+    both that the overall observable is supported, and that its component
+    parts are supported."""
+
+    if isinstance(obs, Tensor):
```

Comment:
Should tensor itself not be checked for support as well?

Reply:
You would think so, but no. Historically everything supports Tensor, so it's actually not included on hard-coded lists of supported observables, and we just check that its components are supported. 

Try `qml.device("lightning.qubit", wires=3).observables`, for example. It does not include `"Tensor"`, but it does still support `Tensor` 🤷‍♀️ 

Hopefully soon it will be gone and we'll only have `Prod` 🙏

Reply:
Huh weird indeed 😅 I feel like it would still be important to check because ultimately what we want to ensure is that a particular device implemented the `__catalyst__qis__TensorObs` (or rather the `QuantumDevice::TensorObservable` method).

It's true that Lightning also doesn't list it in its toml file, but we might want to remedy this in a followup for consistency 🤔 Should we leave a todo?

Reply:
I'm hoping `Tensor` will be removed completely in a couple of releases, so not sure how high a priority it is to update this. But no harm including it as a task if there's time! And maybe `Tensor` will stick around longer than I'd like 😅

Reply:
```suggestion
    # TODO: Check whether Tensor is supported by the device.
    if isinstance(obs, Tensor):
```